### PR TITLE
Add TLS Config to api gateway integration

### LINKED ISSUE #15492: Add TLS_CONFIG block to API Gateway Integration
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

API Gateway Integration supports a TLS Config block that allows you to specify a flag `insecureSkipVerification`.

This functionality is present in the AWS SDK but has not been added to the Terraform AWS provider

See this doc for more info on the API Gateway Integration
https://docs.aws.amazon.com/apigateway/api-reference/resource/integration/

### New or Affected Resource(s)

* aws_api_gateway_integration

### Potential Terraform Configuration

```hcl
resource "aws_api_gateway_integration" "integration" {
    tls_config {
      insecure_skip_verification = true
  }
}
```

### References

* NONE

### PR DESCRIPTION
**Pull request by tmcgahern**
Co-authored-by: jake-mcdermott <mail@jake.ie>

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #15492

The change was based on a similar tls_config block in the API gateway V2 integration here:
https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_apigatewayv2_integration.go#L132

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_api_gateway_integration: Add `tls_config` block to support `insecureSkipVerification` 
```

Output from acceptance testing:

```
make testacc TESTARGS='-run=TestAccAWSAPIGatewayIntegration'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayIntegration -timeout 120m
=== RUN   TestAccAWSAPIGatewayIntegrationResponse_basic
=== PAUSE TestAccAWSAPIGatewayIntegrationResponse_basic
=== RUN   TestAccAWSAPIGatewayIntegrationResponse_disappears
=== PAUSE TestAccAWSAPIGatewayIntegrationResponse_disappears
=== RUN   TestAccAWSAPIGatewayIntegration_basic
=== PAUSE TestAccAWSAPIGatewayIntegration_basic
=== RUN   TestAccAWSAPIGatewayIntegration_contentHandling
=== PAUSE TestAccAWSAPIGatewayIntegration_contentHandling
=== RUN   TestAccAWSAPIGatewayIntegration_cache_key_parameters
=== PAUSE TestAccAWSAPIGatewayIntegration_cache_key_parameters
=== RUN   TestAccAWSAPIGatewayIntegration_integrationType
=== PAUSE TestAccAWSAPIGatewayIntegration_integrationType
=== RUN   TestAccAWSAPIGatewayIntegration_disappears
=== PAUSE TestAccAWSAPIGatewayIntegration_disappears
=== CONT  TestAccAWSAPIGatewayIntegrationResponse_basic
=== CONT  TestAccAWSAPIGatewayIntegration_cache_key_parameters
=== CONT  TestAccAWSAPIGatewayIntegration_disappears
=== CONT  TestAccAWSAPIGatewayIntegration_basic
=== CONT  TestAccAWSAPIGatewayIntegration_contentHandling
=== CONT  TestAccAWSAPIGatewayIntegration_integrationType
=== CONT  TestAccAWSAPIGatewayIntegrationResponse_disappears
=== CONT  TestAccAWSAPIGatewayIntegration_disappears
    resource_aws_api_gateway_integration_test.go:296: [INFO] Got non-empty plan, as expected
=== CONT  TestAccAWSAPIGatewayIntegrationResponse_disappears
    resource_aws_api_gateway_integration_response_test.go:67: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSAPIGatewayIntegration_disappears (128.41s)
--- PASS: TestAccAWSAPIGatewayIntegrationResponse_disappears (132.04s)
--- PASS: TestAccAWSAPIGatewayIntegration_cache_key_parameters (143.51s)
--- PASS: TestAccAWSAPIGatewayIntegrationResponse_basic (220.33s)
2020/10/05 20:01:55 [INFO] Successfully derived credentials from session
2020/10/05 20:01:55 [INFO] AWS Auth provider used: "AssumeRoleProvider"
--- PASS: TestAccAWSAPIGatewayIntegration_contentHandling (282.20s)
--- PASS: TestAccAWSAPIGatewayIntegration_basic (374.41s)
--- PASS: TestAccAWSAPIGatewayIntegration_integrationType (917.51s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       920.219s```

### CODE REVIEW

File: aws/resource_aws_api_gateway_integration_test.go [hunk 1]
```diff
@@ -272,11 +272,10 @@ func TestAccAWSAPIGatewayIntegration_integrationType(t *testing.T) {
 				),
 			},
 			{
-				Config: testAccAWSAPIGatewayIntegrationConfig_IntegrationTypeInternet(rName),
```
Comment (tmcgahern) [inline aws/resource_aws_api_gateway_integration_test.go:275]: This test is the same as [Line 259](https://github.com/terraform-providers/terraform-provider-aws/pull/15499/files#diff-b1909efafe91a419f2290615c3c0e80bR259) so instead of creating another test I've replaced the duplicate with the new test

Comment (bflad) [reply aws/resource_aws_api_gateway_integration_test.go:275]: If I recall correctly, the update back to `INTERNET` was an intentional update test. In general, the maintainers prefer if existing testing is not touched to ensure regressions are covered.

File: aws/resource_aws_api_gateway_integration_test.go [hunk 2]
```diff
@@ -272,11 +272,10 @@ func TestAccAWSAPIGatewayIntegration_integrationType(t *testing.T) {
...
+				Config: testAccAWSAPIGatewayIntegrationConfig_IntegrationTLSConfig(rName),
```
Comment (bflad) [inline aws/resource_aws_api_gateway_integration_test.go:275]: Will move this to its own acceptance test and make the `insecure_skip_verification` a boolean function parameter so we can easily verify updates between true/false.

**Review by bflad — COMMENTED** [review]
Thank you for submitting this, @tmcgahern. Overall it looks really good and we will be pulling this in with just some minor testing updates.
